backend/dataall/modules/shares_base/db/share_object_item_repositories.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from sqlalchemy import and_, or_, func, case
+from sqlalchemy.orm import Query
+from typing import List
+
+from dataall.base.db import exceptions, paginate
+from dataall.base.db.paginator import Page
+from dataall.core.organizations.db.organization_models import Organization
+from dataall.core.environment.db.environment_models import Environment, EnvironmentGroup
+from dataall.modules.datasets_base.db.dataset_models import DatasetBase
+from dataall.modules.datasets_base.db.dataset_repositories import DatasetBaseRepository
+from dataall.modules.notifications.db.notification_models import Notification
+from dataall.modules.shares_base.db.share_object_models import ShareObjectItem, ShareObject, ShareObjectItemDataFilter
+from dataall.modules.shares_base.db.share_state_machines_repositories import ShareStatusRepository
+
+from dataall.modules.shares_base.services.shares_enums import (
+    ShareItemHealthStatus,
+    PrincipalType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ShareObjectItemRepository:
+    @staticmethod
+    def get_share_item_filter_by_uri(session, attached_filter_uri):
+        return (
+            session.query(ShareObjectItemDataFilter)
+            .filter(ShareObjectItemDataFilter.attachedDataFilterUri == attached_filter_uri)
+            .first()
+        )
+
+    @staticmethod
+    def count_all_share_item_filters_with_data_filter_uri(session, filter_uri):
+        return (
+            session.query(ShareObjectItemDataFilter)
+            .filter(
+                ShareObjectItemDataFilter.dataFilterUris.contains(f'{{{filter_uri}}}'),
+            )
+            .count()
+        )
+
+    @staticmethod
+    def get_share_item_by_item_filter_uri(session, uri):
+        return session.query(ShareObjectItem).filter(ShareObjectItem.attachedDataFilterUri == uri).first()
+
+    @staticmethod
+    def delete_share_item_filter(session, share_item_filter) -> bool:
+        session.delete(share_item_filter)
+
+    @staticmethod
+    def update_share_item_filters(
+        session,
+        share_item_filter: ShareObjectItemDataFilter,
+        data: dict,
+    ) -> bool:
+        share_item_filter.label = data.get('label')
+        share_item_filter.dataFilterUris = data.get('filterUris')
+        share_item_filter.dataFilterNames = data.get('filterNames')
+        session.commit()
```

Comment:
In the update, is it possible to not provide some of these keys? For example, let's say we do not update the label, would we update the label to None?

Reply:
a user can try to leave one of those fields empty but label, dataFilterUris, and dataFilterNames are all non-nullable

From FE - we set the initial values for label, dataFilterUris, and dataFilterNames to be the existing values of ShareObjectItemDataFilter (if exist) so users can start from where they left off

If trying to invoke programmatically - one would have to specify all 3 of label, dataFilterUris, and dataFilterNames